pkg/types/baremetal/validation/platform.go

```diff
@@ -32,6 +33,26 @@ func validateIPNotinMachineCIDR(ip string, n *types.Networking) error {
 	return nil
 }
 
+func validateOSImageURI(uri string) error {
+	// Check for valid URI and sha256 checksum part of the URL
+	var sha256Checksum string
```

Comment:
nit: move this to https://github.com/openshift/installer/pull/2757/files#diff-144e602ba4c4174e8b2e21f6d43d59d6R44 closer to the first usage..